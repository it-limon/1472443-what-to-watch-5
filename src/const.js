export const MAX_COMMENTS_PER_COLUMN = 2;
export const MAX_GENRES_COUNT = 9;
export const SIMILAR_MOVIES_COUNT = 4;
export const DEFAULT_SHOWN_MOVIES_COUNT = 8;
export const DEFAULT_REVIEW_RATING = 3;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const ALL_GENRES = `All genres`;

export const MovieRating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very Good`,
  AWESOME: `Awesome`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppPages = {
  MAIN: `MAIN`,
  AUTH: `AUTH`,
  MOVIE: `MOVIE`,
  MYLIST: `MYLIST`,
  REVIEW: `REVIEW`,
  PLAYER: `PLAYER`
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FILMS: `/films`,
  MYLIST: `/mylist`,
  REVIEW: `/review`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  FAVORITE_FILMS: `/favorite`,
  PROMO_MOVIE: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`
};
