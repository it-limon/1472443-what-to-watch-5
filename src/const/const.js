export const MovieRatings = [`Bad`, `Normal`, `Good`, `Very Good`, `Awesome`];
export const REVIEW_COLUMNS_COUNT = 2;
export const LIKE_THIS_MOVIE_COUNT = 4;
