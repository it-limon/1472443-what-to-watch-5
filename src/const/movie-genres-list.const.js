export const MovieGenresList = [
  {
    key: 0,
    ord: 0,
    name: `All genres`,
    filterName: `All genres`,
  },
  {
    key: 1,
    ord: 1,
    name: `Comedie`,
    filterName: `Comedies`,
  },
  {
    key: 2,
    ord: 2,
    name: `Crime`,
    filterName: `Crime`,
  },
  {
    key: 3,
    ord: 3,
    name: `Documentary`,
    filterName: `Documentary`,
  },
  {
    key: 4,
    ord: 4,
    name: `Drama`,
    filterName: `Dramas`,
  },
  {
    key: 5,
    ord: 5,
    name: `Horror`,
    filterName: `Horror`,
  },
  {
    key: 6,
    ord: 6,
    name: `Kids & Family`,
    filterName: `Kids & Family`,
  },
  {
    key: 7,
    ord: 7,
    name: `Romance`,
    filterName: `Romance`,
  },
  {
    key: 8,
    ord: 8,
    name: `Sci-Fi`,
    filterName: `Sci-Fi`,
  },
  {
    key: 9,
    ord: 9,
    name: `Thriller`,
    filterName: `Thrillers`,
  }
];
