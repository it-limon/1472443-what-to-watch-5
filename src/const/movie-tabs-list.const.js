export const MovieTabsList = [
  {
    key: 0,
    name: `Overview`
  },
  {
    key: 1,
    name: `Details`
  },
  {
    key: 2,
    name: `Reviews`
  }
];
