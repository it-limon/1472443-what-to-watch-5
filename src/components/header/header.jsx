import React, {Fragment} from "react";
import PropTypes from "prop-types";
import Props from "../../props";
import {getAuthorizationStatus, getUserInfo} from "../../store/selectors/user-selector";
import {connect} from "react-redux";
import {AuthorizationStatus} from "../../const";
import {Link} from "react-router-dom";

const Header = (props) => {
  const {authorized, userInfo, headerClassName, withActiveMainLink} = props;

  const logoLetter = () => (
    <Fragment>
      <span className="logo__letter logo__letter--1">W</span>
      <span className="logo__letter logo__letter--2">T</span>
      <span className="logo__letter logo__letter--3">W</span>
    </Fragment>
  );

  return (
    <header className={`page-header ${headerClassName}`}>
      <div className="logo">
        <Fragment>
          {withActiveMainLink ?
            <Link className="logo__link" to="/">
              {logoLetter()}
            </Link> :
            <div className="logo__link">
              {logoLetter()}
            </div>
          }
        </Fragment>
      </div>

      <div className="user-block">
        {authorized ?
          <Link to="/mylist">
            <div className="user-block__avatar">
              <img src={userInfo.avatarUrl} alt={userInfo.name} width="63" height="63" />
            </div>
          </Link> :
          <Link to="/login" className="user-block__link">
            Sign In
          </Link>
        }
      </div>
    </header>
  );
};

Header.defaultProps = {
  withActiveMainLink: true
};

Header.propTypes = {
  authorized: PropTypes.bool.isRequired,
  userInfo: Props.userInfo,
  headerClassName: PropTypes.string.isRequired,
  withActiveMainLink: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  authorized: getAuthorizationStatus(state) === AuthorizationStatus.AUTH,
  userInfo: getUserInfo(state)
});

export {Header};
export default connect(mapStateToProps)(Header);
