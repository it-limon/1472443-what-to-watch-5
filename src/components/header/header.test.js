import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header";
import {Route, BrowserRouter} from "react-router-dom";
import {AppPages} from "../../const";

const noAuthUserInfo = {
  id: -1,
  email: ``,
  name: ``,
  avatarUrl: ``
};

const userInfo = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://avatars.dicebear.com/api/human/1.svg`
};

describe(`Render Header`, () => {

  it(`Render Header (authorized/without breadcrumbs/with active user block)`, () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Route>
            <Header
              authorized={true}
              userInfo={userInfo}
              currentPage={AppPages.MAIN}
              breadcrumbsMovieId={-1}
              breadcrumbsMovieName={``}
            />
          </Route>
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render Header (not authorized)`, () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Route>
            <Header
              authorized={false}
              userInfo={noAuthUserInfo}
              currentPage={AppPages.MAIN}
              breadcrumbsMovieId={-1}
              breadcrumbsMovieName={``}
            />
          </Route>
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
