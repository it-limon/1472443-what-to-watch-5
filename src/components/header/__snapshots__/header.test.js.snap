// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Render Header (authorized/without breadcrumbs/with active user block) 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <div
      className="logo__link"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </div>
  </div>
  <div
    className="user-block"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User Name"
          height="63"
          src="https://avatars.dicebear.com/api/human/1.svg"
          width="63"
        />
      </div>
    </a>
  </div>
</header>
`;

exports[`Render Header Render Header (not authorized) 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <div
      className="logo__link"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </div>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign In
    </a>
  </div>
</header>
`;
